pkg/log/zap/zap.go

```diff
@@ -98,6 +100,14 @@ func Encoder(encoder zapcore.Encoder) func(o *Options) {
 	}
 }
 
+// TimeEncoder configures how the logger will encode time format.
+// See Options.TimeEncoder
+func TimeEncoder(timeEncoder string) func(o *Options) {
+	return func(o *Options) {
+		o.TimeEncoder = newTimeEncoder(timeEncoder)
```

Comment:
@bharathi-tenneti Can you test and update this per Joe's suggestion below to use `zapcore.TimeEncoder` as the input arg type.
```suggestion
func TimeEncoder(timeEncoder zapcore.TimeEncoder) func(o *Options) {
	return func(o *Options) {
		o.TimeEncoder = timeEncoder
```
If `zap.New(zap.TimeEncoder(zapcore.EpochMillisTimeEncoder))` works then that's better than parsing a string.

Reply:
It is working as expected. 
Using helper func:
main.go
```go
logger := zap.New(zap.TimeEncoder(zapcore.EpochMillisTimeEncoder), zap.UseDevMode(true))
```
output
------
```shell
$ operator-sdk run --local
INFO[0000] Running the operator locally in namespace default. 
1.5880158036435872e+12	INFO	cmd	Operator Version: 0.0.1
1.588015803643628e+12	INFO	cmd	Go Version: go1.14.2
1.5880158036436328e+12	INFO	cmd	Go OS/Arch: darwin/amd64
1.588015803643636e+12	INFO	cmd	Version of operator-sdk: v0.17.0+git
1.588015803644689e+12	INFO	leader	Trying to become the leader.
1.5880158036447e+12	INFO	leader	Skipping leader election; not running in a cluster.
1.58801580396404e+12	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
1.588015803964294e+12	INFO	cmd	Registering Components.
1.588015803964316e+12	INFO	cmd	Starting the Cmd.
1.588015803964658e+12	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}

```
Using flags:
-----------
```go
logger := zap.New(zap.UseFlagOptions(&opts))
```
```shell
 $ operator-sdk run --local --operator-flags="--zap-encoder=console --zap-time-encoder=nanos"
INFO[0000] Running the operator locally in namespace default. 
1588015875500174000	INFO	cmd	Operator Version: 0.0.1
1588015875500221000	INFO	cmd	Go Version: go1.14.2
1588015875500228000	INFO	cmd	Go OS/Arch: darwin/amd64
1588015875500235000	INFO	cmd	Version of operator-sdk: v0.17.0+git
1588015875501658000	INFO	leader	Trying to become the leader.
1588015875501674000	INFO	leader	Skipping leader election; not running in a cluster.
1588015875812863000	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
1588015875813021000	INFO	cmd	Registering Components.
1588015875813041000	INFO	cmd	Starting the Cmd.
1588015875813202000	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}

```